drivers/serial/Kconfig.native_posix

```diff
@@ -39,6 +39,13 @@ config NATIVE_UART_0_ON_STDINOUT
 
 endchoice
 
+config UART_NATIVE_WAIT_PTS_READY_ENABLE
+	bool "Support waiting for pseudo terminal client readiness"
+	depends on NATIVE_UART_0_ON_OWN_PTY || UART_NATIVE_POSIX_PORT_1_ENABLE
+	help
+	  This option allows for a pts write to wait for the recipient pseudo
```

Comment:
A bit more elaborate description would probably help users and future developers:
```
	  When this option is selected a new command line switch is provided:
	  `--uart-wait`
	  When `--uart-wait` is used, writes to the UART will be held until a client
	  has connected to the slave side of the pseudoterminal.
	  Otherwise writes are sent irrespectively.
```
(see comment about renaming the option below)